crates/autopilot/src/infra/persistence/dto/auction.rs

```diff
@@ -30,7 +34,17 @@ pub fn to_domain(auction: Auction) -> domain::Auction {
             .into_iter()
             .map(super::order::to_domain)
             .collect(),
-        prices: auction.prices,
+        prices: auction
+            .prices
+            .into_iter()
+            .filter_map(|(key, value)| {
+                let price = Price::new(value.into());
+                if let Err(ref e) = price {
```

Comment:
Should I error silently here? or should I change this function from `to_domain()` to `try_to_domain()` and return a `Result<>`?

Reply:
This gets called from the [**function**](https://github.com/cowprotocol/services/blob/bc2ebebfb64eeb62c9219977b595058c5e4dd7cb/crates/autopilot/src/infra/shadow/orderbook/mod.rs#L28) returning `Result` so it might make sense to return a `Result` here as well.

Reply:
I also like more the approach of returning an `Result<>` (it is cleaner).

Reply:
+1 for returning this error instead of creating a potentially inconsistent auction (it doesn't seem as if we can guarantee all prices are non-zero).